platform/feature-api/communication/postoffice/src/rpc.ts

```diff
@@ -28,25 +28,34 @@
  *
  * declare serverPort: ResponsePort<EndpointRequest, EndpointResponse>;
  *
- * server(serverPort, endpointServer(specImpl));
+ * server(serverPort, backendServer(specImpl));
  *
- * declare clientPort: RequestPort<EndpointRequest, EndpointResponse>;
+ * declare clientPort: RequestPort<BackendRequest, BackendResponse>;
  *
- * const rpcClient: ClientOf<Spec> = endpointClient(client(clientPort));
+ * const rpcClient: ClientOf<Spec> = backendClient(client(clientPort));
  * ```
  *
  * @packageDocumentation
  */
 
-import { EndpointSpec, ServerOf, ClientOf, Callable } from "./types";
-import { EndpointProcedure, EndpointRequestPart, EndpointRequest } from "./protocol";
+import { BackendSpec, ServerOf, ClientOf, Callable } from "./types";
+import {
+    BackendProcedure,
+    BackendRequestPart,
+    BackendRequest,
+} from "./protocol";
 
 /**
- * Turns the implementation of an endpoint specification into a plain function.
+ * Turns the implementation of a backend endpoint specification into a plain function.
  */
-export function endpointServer<Spec extends EndpointSpec>(impl: ServerOf<Spec>): EndpointProcedure {
+export function backendServer<Spec extends BackendSpec>(
+    impl: ServerOf<Spec>
+): BackendProcedure {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async function process(impl: any, parts: ReadonlyArray<EndpointRequestPart>): Promise<any> {
+    async function process(
+        impl: any,
+        parts: ReadonlyArray<BackendRequestPart>
```

Comment:
An array of RequestParts is a Request. Not clear why you use a ReadonlyArray here, and not on the definition. At any rate, this seems to be functionally equivalent.

Reply:
unclear about that as well, maybe they don't want the array to be mutable?

Reply:
But in that case it would be convenient to not make it mutable in any place it appears, right?